cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/partitioned/DynamicPartitioningOutputCommitter.java

```diff
@@ -194,12 +196,30 @@ public void abortJob(JobContext context, JobStatus.State state) throws IOExcepti
     // We want to rollback these partitions.
     if (outputDataset != null) {
       try {
-        ((TransactionAware) outputDataset).rollbackTx();
-      } catch (Throwable t) {
-        LOG.warn("Attempt to rollback partitions failed", t);
+        try {
+          ((TransactionAware) outputDataset).rollbackTx();
+        } catch (Throwable t) {
+          LOG.warn("Attempt to rollback partitions failed", t);
+        }
+
+        // if this is non-null, then at least some paths have been created. We need to remove them
+        if (relativePaths != null) {
```

Comment:
I see... we write all the files, before adding any of the partitions. So if the first partition fails to add, all of the files wont be rolled back.